pkg/operator/resource/resourceapply/core.go

```diff
@@ -196,6 +196,10 @@ func ApplyConfigMap(client coreclientv1.ConfigMapsGetter, recorder events.Record
 
 // ApplySecret merges objectmeta, requires data
 func ApplySecret(client coreclientv1.SecretsGetter, recorder events.Recorder, required *corev1.Secret) (*corev1.Secret, bool, error) {
+	if len(required.StringData) > 0 {
+		return nil, false, fmt.Errorf("Secret.stringData is not supported")
```

Comment:
why not?

Reply:
Because it's normalized by the server to `Data`, and we would have to add that normalization here before the deepcopy. Not impossible, just not done.